sys/include/ztimer/xtimer_compat.h

```diff
@@ -71,7 +71,20 @@ static inline uint64_t xtimer_now_usec64(void)
 static inline void xtimer_sleep(uint32_t seconds)
 {
     /* TODO: use ZTIMER_SEC */
-    ztimer_sleep(ZTIMER_USEC, seconds * 1000000LU);
+    if (IS_ACTIVE(MODULE_ZTIMER_MSEC)) {
```

Comment:
unfortunately, these have overflow issues, because it is possible to pass seconds larger than `(2**32-1)/1000` or `(2**32-1) / 1000000` respectively.

A workaround for `xtimer_msleep()` would be to just make ztimer_xtimer_compat depend on ZTIMER_MSEC, and let ztimer handle it.

`xtimer_sleep()` could do something like:

```
while (seconds > (UINT32_MAX / 1000)) {
xtimer_msleep(UINT32_MAX);
seconds -= UINT32_MAX / 1000;
}
xtimer_msleep(seconds * 1000LU);
```

This would be some overhead for the rare case that someone intents to sleep for 4 million seconds (49 days), but you never know. :)

Reply:
This just might add a second every 49 days and cause an extra wakeup, but both are IMO acceptable.

Reply:
Isn't that already an issue for the `xtimer_sleep()` -> `ztimer_sleep(ZTIMER_USEC, seconds * 1000000LU);` mapping?

Reply:
yes :/